lib/elixlsx/xml_templates.ex

```diff
@@ -432,6 +432,96 @@ defmodule Elixlsx.XMLTemplates do
     end
   end
 
+  defp make_autofilter(nil, filter_cols) do
+    ""
+  end
+
+  defp make_autofilter(autofilter_ref, filter_cols) do
+    {row1, col1, row2, col2} = autofilter_ref
+
+    """
+      <autoFilter ref="#{Util.to_excel_coords(row1, col1)}:#{Util.to_excel_coords(row2, col2)}">
+    """ <>
+    make_filter_columns(autofilter_ref, filter_cols) <>
+    """
+      </autoFilter>
+    """
+  end
+
+  defp make_filter_columns(autofilter_ref, filter_cols) do
+    {_row1, col1, _row2, col2} = autofilter_ref
+
+    Enum.map_join(filter_cols, fn col, {filter_type, filters} ->
+      if col < col1 or col > col2 do
+        # Filter column out of autofilter range
+        ""
+      else
+        # Filter columns are relative to first column in the autofilter.
+        make_filter_column(col - col1, filter_type, filters)
+      end
+    end)
+  end
+
+  defp make_filter_column(col_id, filter_type, filters) do
+    """
+    <filterColumn colId="#{col_id}">
+    """ <>
+    if filter_type == :list, do: make_filters(filters), else: make_custom_filters(filters) <>
+    """
+    </filterColumn>
+    """
+  end
+
+  defp make_filters(filters) do
+    non_blanks = Enum.reject(filters, fn filter -> filter != :blank end)
+
+    blank = length(filters) != length(non_blanks)
+    """
+    <filters blank="#{blank}">
+    """ <>
+    Enum.map_join(non_blanks, &make_filter/1)
+    """
+    </filters>
+    """
+  end
+
+  defp make_filter({%DateTime{year: year, month: month, day: day, hour: hour, minute: minute, second: second} = date, grouping}) do
+    """
+    <dateGroupItem year="#{year}" month="#{month}" day="#{day}" hour="#{hour}" minute="#{minute}" second="#{second}" dateTimeGrouping="#{grouping}"/>
+    """
+  end
+
+  defp make_filter(val) do
+    "<filter val=\"#{val}\"/>"
+  end
+
+  def make_custom_filters(expr) do
```

Comment:
should this be defp?

Reply:
It most definitely should. I have fixed it now.